### PR TITLE
Add transaction building from the keychain

### LINKED ISSUE #40: Add transaction building from the keychain
### Problem to solve

Currently, when we are building a transaction, we are passing the seed and the index in the `TransactionBuilder.build` function.
However, with the keychain with can't leverage this function, so for this reason and hardware reason a function have been implemented to set the signature and public key of the transaction and not using the `build` function.
So we can still use the `setPreviousSignatureAndPreviousPublicKey` and using `keychain.deriveKeypair` to  leverage the tx builder and using the `Crypto.sign` function.
But this is not really easy or friendly, as it implies multiple steps.
So we should provide a better API to build transaction from the keychain services.

### Solution

The idea of the proposal would be to add a function in the keychain to build a transaction from a `TransactionBuilder` instance and abstract all those functions calls into a single one.

For example, we may have a function like: `keychain.buildTransaction(txBuilder, service, index)` where we are passing the builder as argument as well the service and the index of the keys to generate. Then the keys and signature will be generated and pass to the builder through the functions listed above. 
This way from a keychain's perspective it makes transaction building easier for devs and users of the lib.

### PR DESCRIPTION
**Pull request by Neylix**
Fixes #40 
Add new `keychain.buildTransaction` function.
Add test for this function as for txBuilder
Add UCO transfert feature in example keychain.html

Tested with the example and the test file

### CODE REVIEW

File: lib/keychain.js [hunk 1]
```diff
@@ -76,6 +84,18 @@ class Keychain {
...
+  buildTransaction(transaction, service, index) {
+    const keypair = this.deriveKeypair(service, index)
+    const address = this.deriveAddress(service, index + 1)
+
+    transaction.setAddress(address)
+
+    const payloadForPreviousSignature = transaction.previousSignaturePayload()
+    const previousSignature = sign(payloadForPreviousSignature, keypair.privateKey)
+    
+    transaction.setPreviousSignatureAndPreviousPublicKey(previousSignature, keypair.publicKey)
```
Comment (samuel-uniris) [inline lib/keychain.js:96]: WIll be great to return the transaction builder to avoid any confusion (FP style)

File: lib/api.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ module.exports = {
             handler(result.data.transactionConfirmed.nbConfirmations)
           }
 
-          throw result
+          resolve(result)
```
Comment (samuel-uniris) [inline lib/api.js:107]: Why did you change this ?

**Review by samuel-uniris — CHANGES_REQUESTED** [review]
Good PR
Few comments.
If you can also add instruction in the README

File: lib/api.js (hunk 2 above)
Comment (Neylix) [reply lib/api.js]: And removed

**Review by samuel-uniris — APPROVED** [review]
